sdk/devcenter/developer-devcenter-rest/test/public/utils/recordedClient.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Context } from "mocha";
+import { isPlaybackMode, Recorder, RecorderStartOptions } from "@azure-tools/test-recorder";
+import { createTestCredential } from "@azure-tools/test-credential";
+import { DefaultAzureCredential } from "@azure/identity";
+import createClient from "../../../src/azureDevCenter";
+
+import "./env";
+import { ClientOptions } from "@azure-rest/core-client";
+
+const envSetupForPlayback: Record<string, string> = {
+  ENDPOINT: "https://endpoint",
+  AZURE_CLIENT_ID: "azure_client_id",
+  AZURE_CLIENT_SECRET: "azure_client_secret",
+  AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
+  DEVCENTER_TENANT_ID: "88888888-8888-8888-8888-888888888888",
+  DEFAULT_DEVCENTER_NAME: "sdk-test-devcenter",
+  SUBSCRIPTION_ID: "azure_subscription_id",
+};
+
+const recorderEnvSetup: RecorderStartOptions = {
+  envSetupForPlayback,
+};
+
+/**
+ * creates the recorder and reads the environment variables from the `.env` file.
+ * Should be called first in the test suite to make sure environment variables are
+ * read before they are being used.
+ */
+export async function createRecorder(context: Context): Promise<Recorder> {
+  const recorder = new Recorder(context.currentTest);
+  await recorder.start(recorderEnvSetup);
+  return recorder;
+}
+
+export function createRecordedClient(
+  recorder: Recorder,
+  tenantId: string,
+  devCenter: string,
+  options: ClientOptions = {},
+  devCenterDnsSuffix: string = "devcenter.azure.com"
+) {
```

Comment:
```suggestion
): AzureDevCenterClient {
```

To bypass the linter check issue https://dev.azure.com/azure-sdk/public/_build/results?buildId=1925208&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=c9190431-7417-53a5-3fc1-7c7e026cd79d&l=124.